internal/controller/storageclaim_controller.go

```diff
@@ -397,6 +398,10 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 					storageClass = r.getCephFSStorageClass(data)
 				} else if resource.Name == "ceph-rbd" {
 					storageClass = r.getCephRBDStorageClass(data)
+
+					if enable, ok := r.storageClaim.GetAnnotations()[keyRotationEnableAnnotation]; ok && enable != "true" {
```

Comment:
does this ann. value supports anything other than `true/false`? why to check for a negation, like why not `ann[key] == "false" ? set-false`

if the value is "true"/"false" then it can be directly used, isn't it?
`utils.AddAnnotation(storageClass, keyRotationEnableAnnotation, r.storageClaim.GetAnnotations()[keyRotationEnableAnnotation])`

Reply:
No it does not support any other value than true/false, values other than `true` are considered false. I have no good reason on why it cannot be used directly (as it will be parsed in csi-addons anyways)

The reason I did not do it such way is to avoid setting it unconditionally because in `utils.AddAnnotation` we set the `value` even if it is empty. Want me to change it?

Also, while we are at it, should I update the code to update the SC by patching it instead of deleting and recreating in case the annotations differ? Would be a little efficient IG?

Reply:
> Want me to change it?
- No, annotation w/ an empty value is also a valid identifier, let's not change it

> Also, while we are at it, should I update the code to update the SC by patching it instead of deleting and recreating in case the annotations differ? Would be a little efficient IG?
- these values converge easily in a couple of reconciles and a patch here makes less difference. 
- Nevertheless will revisit when VolumeAttributes (Madhu mentioned it once) feature which helps in storageclass having almost static data.

Reply:
> * No, annotation w/ an empty value is also a valid identifier, let's not change it
> 
> >

I meant changing the current pr aka my change-set to add annotation unconditionally :)

I am still leaning towards adding the annotation only when its value is not truthy.

Reply:
> * these values converge easily in a couple of reconciles and a patch here makes less difference.

Would you please explain it to me? Acc to my understanding we would be deleting and recreating the SC in case annotations differ. That is two different actions while patching the annotations on an existing SC will not disrupt any operations and is one shot.

Reply:
> in case annotations differ
- yes, I mean these doesn't change often, isn't it.

Reply:
Correct.